### PR TITLE
Fix tab completion and toml reading

### LINKED ISSUE #574: Hitting TAB on package with conflicts leads to error
In the following log, on `0.7.0-rc2`, I was trying to figure out what to do with the `Tb5classes` requirements. I wanted to try `up Tb5classes`, i.e. I started with `up Tb5` and hit the tab key - resulting in the keymap error.

It seems that to reproduce one needs a conflicting packages state, i.e. where `up` yields an error, then run `up SOMECHARS`+tab to trigger the bug. `SOMECHARS` does not have to be the beginning of a package name, it can be anything.
 
```jl
  Updating git-repo `https://github.com/fredrikekre/JLD2.jl.git`
  Updating git-repo `https://github.com/rened/NIfTI.jl.git`
 Resolving package versions...
ERROR: Unsatisfiable requirements detected for package Tb5classes [c0dbe194]:
 Tb5classes [c0dbe194] log:
 ├─Tb5classes [c0dbe194] has no known versions!
 └─restricted to versions * by Visceral [ffac7baf] — no versions left
   └─Visceral [ffac7baf] log:
     ├─possible versions are: 0.1.0 or uninstalled
     └─Visceral [ffac7baf] is fixed to version 0.1.0

(REXBackend) pkg> up Tb5┌ Error: Error in the keymap
│   exception =
│    type Nothing has no field ver
│    Stacktrace:
│     [1] getproperty(::Any, ::Symbol) at ./sysimg.jl:18
│     [2] installed(::Pkg.Types.PackageMode) at /home/rene/local/julia/usr/share/julia/stdlib/v0.7/Pkg/src/API.jl:249
│     [3] complete_installed_package at /home/rene/local/julia/usr/share/julia/stdlib/v0.7/Pkg/src/REPLMode.jl:924 [inlined]
│     [4] complete_package at /home/rene/local/julia/usr/share/julia/stdlib/v0.7/Pkg/src/REPLMode.jl:916 [inlined]
│     [5] completions(::String, ::Int64) at /home/rene/local/julia/usr/share/julia/stdlib/v0.7/Pkg/src/REPLMode.jl:1002
│     [6] complete_line(::Pkg.REPLMode.PkgCompletionProvider, ::REPL.LineEdit.PromptState) at /home/rene/local/julia/usr/share/julia/stdlib/v0.7/Pkg/src/REPLMode.jl:896
│     [7] complete_line(::REPL.LineEdit.PromptState, ::Any) at /home/rene/local/julia/usr/share/julia/stdlib/v0.7/REPL/src/LineEdit.jl:317
│     [8] complete_line(::REPL.LineEdit.MIState) at /home/rene/local/julia/usr/share/julia/stdlib/v0.7/REPL/src/LineEdit.jl:308
│     [9] edit_tab(::REPL.LineEdit.MIState, ::Any, ::Any) at /home/rene/local/julia/usr/share/julia/stdlib/v0.7/REPL/src/LineEdit.jl:1942 (repeats 2 times)
│     [10] (::getfield(REPL.LineEdit, Symbol("##108#160")))(::Any, ::Any, ::Vararg{Any,N} where N) at /home/rene/local/julia/usr/share/julia/stdlib/v0.7/REPL/src/LineEdit.jl:1983
│     [11] #invokelatest#1 at ./essentials.jl:691 [inlined]
│     [12] invokelatest at ./essentials.jl:690 [inlined]
│     [13] (::getfield(REPL.LineEdit, Symbol("##27#28")){getfield(REPL.LineEdit, Symbol("##108#160")),String})(::Any, ::Any) at /home/rene/local/julia/usr/share/julia/stdlib/v0.7/REPL/src/LineEdit.jl:1319
│     [14] prompt!(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /home/rene/local/julia/usr/share/julia/stdlib/v0.7/REPL/src/LineEdit.jl:2353
│     [15] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /home/rene/local/julia/usr/share/julia/stdlib/v0.7/REPL/src/LineEdit.jl:2256
│     [16] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at /home/rene/local/julia/usr/share/julia/stdlib/v0.7/REPL/src/REPL.jl:1029
│     [17] run_repl(::REPL.AbstractREPL, ::Any) at /home/rene/local/julia/usr/share/julia/stdlib/v0.7/REPL/src/REPL.jl:191
│     [18] (::getfield(Base, Symbol("##831#833")){Bool,Bool,Bool,Bool})(::Module) at ./logging.jl:311
│     [19] #invokelatest#1 at ./essentials.jl:691 [inlined]
│     [20] invokelatest at ./essentials.jl:690 [inlined]
│     [21] macro expansion at ./logging.jl:308 [inlined]
│     [22] run_main_repl(::Bool, ::Bool, ::Bool, ::Bool, ::Bool) at ./client.jl:340
│     [23] exec_options(::Base.JLOptions) at ./client.jl:252
│     [24] _start() at ./client.jl:432
└ @ REPL.LineEdit ~/local/julia/usr/share/julia/stdlib/v0.7/REPL/src/LineEdit.jl:2355
(REXBackend) pkg>```

### PR DESCRIPTION
**Pull request by 00vareladavid**
This should fix #574.

1. We can't help if users break their Manifest/Project files. So just refuse to complete if anything goes wrong when reading project/manifest files.
2. Apparently, `TOML` throws a composite exception (whatever that is) so the previous way of checking (with `err isa TOML.ParserError`) was not catching parse errors.

### CODE REVIEW

**Comment by 00vareladavid** [comment]
bors try

File: src/Types.jl [hunk 1]
```diff
@@ -532,26 +532,24 @@ function Project(raw::Dict)
...
+        else
+            rethrow()
+        end
     end
-    return raw, err
-end
-
-function read_project(io::IO)
-    raw, err = _read_project(io)
-    err === nothing ? Project(raw) : pkgerror("Could not parse project file: $(err.msg)")
```
[the remark this answers is not in the document]
Comment (00vareladavid) [reply src/Types.jl]: That error is still there. I think you are talking about printing the filename? If so, I agree.

[the remark this answers is not in the document]
Comment (fredrikekre) [reply src/Types.jl]: Yea, the filename.

[the remark this answers is not in the document]
Comment (00vareladavid) [reply src/Types.jl]: K, looks good now?

File: src/REPLMode.jl [hunk 2]
```diff
@@ -787,11 +787,20 @@ function canonical_names()
...
+        unique!([entry.name for (uuid, entry) in env.manifest])
+end
+
 function complete_installed_packages(options, partial)
-    mode = get(options, :mode, nothing)
-    pkgs = mode == PKGMODE_MANIFEST ? API.__installed() : API.__installed(PKGMODE_PROJECT)
-    stdlib_names = collect(values(Types.stdlib()))
-    return collect(keys(filter(p -> p[1] in stdlib_names || p[2] !== nothing, pkgs)))
+    mode = get(options, :mode, PKGMODE_PROJECT)
+    return installed_names(mode == PKGMODE_PROJECT)
```
Comment (fredrikekre) [inline src/REPLMode.jl]: `installed_names` seems unecessary since its not used anywhere else?

Comment (00vareladavid) [reply src/REPLMode.jl]: Agreed.

**Pull request merged**